core/src/utils/overlays.ts

```diff
@@ -669,7 +669,30 @@ export const createDelegateController = (ref: {
     return { inline, delegate: workingDelegate };
   };
 
+  /**
+   * Attaches a component in the DOM. Teleports the component
+   * to the root of the app.
+   * @param component The component to optionally construct and append to the element.
+   */
+  const attachViewToDom = async (component?: any) => {
+    const { delegate } = getDelegate(true);
+    if (delegate) {
+      await delegate.attachViewToDom(ref.el, component);
+    }
```

Comment:
Should this throw an error if there's no delegate? We have something similar for modal/popover delegates: https://github.com/ionic-team/ionic-framework/blob/main/core/src/utils/framework-delegate.ts#L17